utils/AnalysisUtils.py

```diff
@@ -581,6 +581,44 @@ def ApplySplineFuncToColumn(df, column, spline, minRange=-1.e10, maxRange=1.e10)
     return y
 
 
+def ApplyHistoEntriesToColumn(df, column, histo):
+    '''
+    Method to apply a function to a pandas column via a spline object
+
+    Parameters
+    ----------
+
+    - df: input pandas.Dataframe
+    - column: column of the pandas dataframe to which apply the spline
+    - histo: ROOT.TH1 with values
+
+    Returns
+    ----------
+
+    - y: pandas.Series with result of the function application to column
+
+    '''
+    binMins, binMaxs, contents, y = ([] for _ in range(4))
+    for iBin in range(1, histo.GetNbinsX()+1):
+        binMins.append(histo.GetBinLowEdge(iBin))
+        binMaxs.append(histo.GetBinLowEdge(iBin)+histo.GetBinWidth(iBin))
+        contents.append(histo.GetBinContent(iBin))
+
+    for ix, x in enumerate(df[column].values):
```

Comment:
`ix` is not used and the enumerate is not necessary. It is not possible to avoid the loop on the pandas rows? This is usually very slow.

Reply:
Ciao @fcatalan, I removed the enumerate and `ix` which indeed are not needed (probably they were a leftover of a previous attempt) in this commit a282c84. On the loop on pandas rows, I also tried with [pandas.DataFrame.apply](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.apply.html), by defining a lambda function that returned different values (bin content of the histogram) depending on the pT interval (based on the binning of the histogram) but it was by far slower, so the only solution I could find was the loop over the rows (which is quite fast after all given that the dataframes are of MC signal only). However, if you have better solutions I can implement it. Thanks!

Reply:
Hi @fgrosa, my idea was to try with an apply. But if it is slower, this solution is fine.